### PR TITLE
Added support for postgresql_privs

### PR DESCRIPTION
**Pull request by ArmiT**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -38,6 +38,9 @@ postgresql_users: []
 # List of user privileges to be applied (optional)
 postgresql_user_privileges: []
 
+# List of privileges to be applied (optional)
+postgresql_privileges: []
```
Comment (farridav) [inline defaults/main.yml]: as to avoid being misleading perhaps this should be `postgresql_db_priveleges`?

File: tasks/main.yml [hunk 2]
```diff
@@ -29,6 +29,9 @@
 - include: users_privileges.yml
```
Comment (farridav) [inline tasks/main.yml]: these two can be merged together into priveleges.yml...

File: tasks/privileges.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    password: "{{ item.password | default(\"\") }}"
+    privs: "{{ item.privs | default(\"ALL\") }}"
+    roles: "{{ item.roles | default(\"PUBLIC\") }}"
+    schema: "{{ item.schema | default(omit) }}"
+    state: "{{ item.state | default(\"present\") }}"
+    type: "{{ item.type | default(\"table\") }}"
+  sudo: yes
+  sudo_user: "{{postgresql_admin_user}}"
+  with_items: postgresql_privileges
+  when: postgresql_privileges | length > 0
```
Comment (farridav) [inline tasks/privileges.yml]: I don't think this line is needed :)

**Comment by lhoss** [comment]
@ArmiT possible to get conflicts resolved with latest master?
with the hope we can get this merged

**Comment by gclough** [comment]
Is this the same as #181?  Which one should be merged?

**Comment by jlozadad** [comment]
probably this one but, is old and still needs work.

File: defaults/main.yml [hunk 4]
```diff
@@ -38,6 +38,9 @@ postgresql_users: []
 # List of user privileges to be applied (optional)
 postgresql_user_privileges: []
 
+# List of database privileges to be applied (optional)
+postgresql_db_priveleges: []
```
Comment (nchudleigh) [inline defaults/main.yml]: Typo
```suggestion
postgresql_db_privileges: []
```

File: tasks/privileges.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    objs: "{{ item.objs | default(omit) }}"
+    password: "{{ item.password | default(\"\") }}"
+    privs: "{{ item.privs | default(\"ALL\") }}"
+    roles: "{{ item.roles | default(\"PUBLIC\") }}"
+    schema: "{{ item.schema | default(omit) }}"
+    state: "{{ item.state | default(\"present\") }}"
+    type: "{{ item.type | default(\"table\") }}"
+  sudo: yes
+  sudo_user: "{{postgresql_admin_user}}"
+  with_items: postgresql_db_priveleges
```
Comment (nchudleigh) [inline tasks/privileges.yml]: Typo 
```suggestion
  with_items: postgresql_db_privileges
```

File: tests/vars.yml [hunk 6]
```diff
@@ -17,3 +17,11 @@ postgresql_users:
 postgresql_user_privileges:
   - name: baz
     db: foobar
+
+postgresql_db_priveleges:
```
Comment (nchudleigh) [inline tests/vars.yml]: Typo
```suggestion
postgresql_db_privileges:
```

**Comment by nchudleigh** [comment]
I think all the naming here should really be simplified down to `privs` given that the underlying module is called `privs`.

https://docs.ansible.com/ansible/latest/modules/postgresql_privs_module.html

**Comment by nchudleigh** [comment]
@gclough @jlozadad If I open another PR consolidating #181 and #134 with conflicts and naming issues resolved can I get a review sometime soon?

**Comment by gclough** [comment]
@nchudleigh , in short, yes.  I've got some time to help with this project, and have already cleaned up a lot of old PR's.  There are also some new helpers that can review/comment.

**Comment by RCM7** [comment]
@nchudleigh did you end up opening another PR? I'm looking forward to having this. Can also try to contribute if needed.

**Comment by nchudleigh** [comment]
@RCM7 No, I got frustrated with the pace of the maintainers and stopped contributing to this project.  I went with an approach of using the core ansible postgres modules and honestly I am dubious of how much value this project adds.

**Comment by RCM7** [comment]
@nchudleigh yeah I ended up doing the same !

**Comment by egmont1227** [comment]
Related to Issue #175

**Comment by Pyton** [comment]
Any progress on this?

**Pull request closed without merging**